### ISSUE TITLE
Broken integer rounding for `big(Inf)`

### ISSUE THREAD

**Issue by Keno** [JuliaLang/julia#44662]
```
julia> round(Integer, Inf)
ERROR: InexactError: trunc(Int64, Inf)
Stacktrace:
 [1] trunc
   @ ./float.jl:781 [inlined]
 [2] trunc
   @ ./float.jl:354 [inlined]
 [3] round(#unused#::Type{Integer}, x::Float64)
   @ Base ./float.jl:359
 [4] top-level scope
   @ REPL[14]:1

julia> round(Integer, big(Inf))
0
```

**Comment by barucden** [comment]
Also:
```
julia> round(Integer, NaN)
ERROR: InexactError: trunc(Int64, NaN)
Stacktrace:
 [1] trunc
   @ ./float.jl:805 [inlined]
 [2] trunc
   @ ./float.jl:364 [inlined]
 [3] round(#unused#::Type{Integer}, x::Float64)
   @ Base ./float.jl:369
 [4] top-level scope
   @ REPL[16]:1

julia> round(Integer, big(NaN))
0
```
I think it is due to [mpfr.jl#L280](https://github.com/JuliaLang/julia/blob/v1.7.2/base/mpfr.jl#L280) calling `mpfr_get_z` which returns 0 in case of `NaN` or `Inf`:
>If op is NaN or an infinity, the erange flag is set, rop is set to 0, and 0 is returned.

Is `_unchecked_cast` the correct place to check for `NaN` or `Inf`?

**Comment by Keno** [comment]
The other method doesn't quite work for BigInt, because
```
julia> typemin(BigInt)
ERROR: MethodError: no method matching typemin(::Type{BigInt})
```
but yes, add the same logic for checking the erange flag.

**Comment by barucden** [comment]
Solved by #44676 (I just messed up the "fixes" keyword in the PR message)
